js/bibox.js

```diff
@@ -677,23 +677,50 @@ module.exports = class bibox extends Exchange {
         /**
          * @method
          * @name bibox#fetchOrderBook
+         * @see https://biboxcom.github.io/api/spot/v4/en/#get-order-book
          * @description fetches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
          * @param {string} symbol unified symbol of the market to fetch the order book for
-         * @param {int|undefined} limit the maximum amount of order book entries to return
+         * @param {int|undefined} limit *default=100* valid values include 1, 2, 5, 10, 20, 50, 100, 200, 500, 1000
          * @param {object} params extra parameters specific to the bibox api endpoint
+         *
+         * EXCHANGE SPECIFIC PARAMETERS
+         * @param {int|undefined} price_scale *default=0* depth of consolidation by price, valid values include 0, 1, 2, 3, 4, 5
          * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-book-structure} indexed by market symbols
          */
         await this.loadMarkets ();
         const market = this.market (symbol);
         const request = {
-            'cmd': 'depth',
-            'pair': market['id'],
+            'symbol': market['id'],
         };
         if (limit !== undefined) {
-            request['size'] = limit; // default = 200
+            const allowedValues = [ 1, 2, 5, 10, 20, 50, 100, 200, 500, 1000 ];
+            if (!this.inArray (limit, allowedValues)) {
+                throw new BadRequest (this.id + ' fetchOrderBook limit argument by only be one of 1, 2, 5, 10, 20, 50, 100, 200, 500 or 1000');
+            }
+            request['level'] = limit;
         }
-        const response = await this.v1PublicGetMdata (this.extend (request, params));
-        return this.parseOrderBook (response['result'], market['symbol'], this.safeNumber (response['result'], 'update_time'), 'bids', 'asks', 'price', 'volume');
+        const response = await this.v4PublicGetMarketdataOrderBook (this.extend (request, params));
+        //
+        //    {
+        //        i: '1917961902',                  // update id
+        //        t: '1666221729812',               // update time
+        //        b: [                              // buy orders
+        //            [
+        //                '0.350983',               // order price
+        //                '8760.69'                 // order amount
+        //            ],
+        //            ...
+        //        ],
+        //        a: [                              // sell orders
+        //            [
+        //                '0.351084',
+        //                '14241.62'
+        //            ],
+        //            ...
+        //        ]
+        //    }
+        //
+        return this.parseOrderBook (response, market['symbol'], this.safeNumber (response, 't'), 'b', 'a');
```

Comment:
@samgermain I'll fix it this time, but we should not parse timestamps as floats, next time, please use `safeInteger`